packages/api/lambdas/sqs-message-consumer.js

```diff
@@ -18,66 +18,95 @@ const Rule = require('../models/rules');
+  const rulesModel = new Rule();
+  const rules = await rulesModel.queryRules({
+    type: 'sqs',
+    state: 'ENABLED'
+  });
 
   const messageLimit = event.messageLimit || 1;
   const timeLimit = event.timeLimit || 240;
 
-  await Promise.all(rules.map(async (rule) => {
+  const rulesByQueueMap = rules.reduce((map, rule) => {
     const queueUrl = rule.rule.value;
+    // eslint-disable-next-line no-param-reassign
+    map[queueUrl] = map[queueUrl] || [];
+    map[queueUrl].push(rule);
+    return map;
+  }, {});
+
+  await Promise.all(Object.keys(rulesByQueueMap).map(async (queueUrl) => {
+    const rulesForQueue = rulesByQueueMap[queueUrl];
 
-    if (!(await sqsQueueExists(queueUrl))) return Promise.resolve();
+    if (!(await sqsQueueExists(queueUrl))) {
+      const ruleIds = rulesForQueue.map((rule) => rule.id);
+      log.info(`Could not find queue ${queueUrl}. Unable to process rules ${ruleIds}`);
+      return Promise.resolve();
+    }
+
+    // Use the max of the visibility timeouts for all the rules
+    // bound to this queue.
+    const visibilityTimeout = rulesHelpers.getMaxTimeoutForRules(rulesForQueue);
 
     const consumer = new Consumer({
-      queueUrl: queueUrl,
+      queueUrl,
       messageLimit,
       timeLimit,
-      visibilityTimeout: rule.meta.visibilityTimeout,
+      visibilityTimeout,
       deleteProcessedMessage: false
     });
-    log.info(`processQueues for rule ${rule.name} and queue ${queueUrl}`);
-    return consumer.consume(dispatchFn.bind({ rule: rule }));
+    log.info(`processing queue ${queueUrl}`);
+
+    return consumer.consume(dispatchFn.bind({
+      queueUrl,
+      rulesForQueue
+    }));
   }));
 }
 
 /**
- * process an SQS message
+ * Process an SQS message
  *
  * @param {Object} message - incoming queue message
  * @returns {Promise} - promise resolved when the message is dispatched
- *
  */
 function dispatch(message) {
-  const queueUrl = this.rule.rule.value;
   const messageReceiveCount = parseInt(message.Attributes.ApproximateReceiveCount, 10);
+  const queueUrl = this.queueUrl;
+  const rulesForQueue = this.rulesForQueue;
 
-  if (get(this.rule, 'meta.retries', 3) < messageReceiveCount - 1) {
-    log.debug(`message ${message.MessageId} from queue ${queueUrl} has been processed ${messageReceiveCount - 1} times, no more retries`);
-    // update visibilityTimeout to 5s
-    const params = {
-      QueueUrl: queueUrl,
-      ReceiptHandle: message.ReceiptHandle,
-      VisibilityTimeout: 5
-    };
-    return sqs().changeMessageVisibility(params).promise();
-  }
+  const eventObject = JSON.parse(message.Body);
+  const eventCollection = rulesHelpers.lookupCollectionInEvent(eventObject);
 
-  if (messageReceiveCount !== 1) {
-    log.debug(`message ${message.MessageId} from queue ${queueUrl} is being processed ${messageReceiveCount} times`);
-  }
+  const rulesToSchedule = rulesHelpers.filterRulesbyCollection(rulesForQueue, eventCollection);
```

Comment:
What if there is no collection in the event?

Reply:
If there is no collection in the event, then `lookupCollectionInEvent` returns an empty object:

https://github.com/nasa/cumulus/pull/1638/files?file-filters%5B%5D=.js#diff-023361c061b8df15ce4e31d181c7e2adR229

And if `filterRulesbyCollection` receives an empty object for `eventCollection`, it will simply return all the rules passed into, so effectively no filtering by collection will happen and workflows will queued for all rules. I added an explicit test for this:

https://github.com/nasa/cumulus/pull/1638/files?file-filters%5B%5D=.js#diff-82db426278e52d5ff0bfdbad86db20ebR282